### PR TITLE
Interactive API beta

### LINKED ISSUE #13: Redesigning Task Runner
- [ ] Not require explicitly setting TensorKeys to track.
- [ ] PyTorch/TensorFlow code can be imported directly into fx.
Define easy to use python API:
     - [x]	Model Provider
     - [x]	Task registering
     - [ ]	Data repository adapter
- [x]	Implement example new task runner in Jupyter notebook
- [x] Build compatibility with new Data Loader interface
Framework adapters for all currently supported frameworks
     - [x] Tensorflow 2
     - [x] PyTorch
- [x] Well defined Interface for user contributed plugins
- [ ] Extensions to CoreTaskRunner
- [ ] Custom task types (beyond ‘train’, ‘validate’)
- [ ] Allow custom function arguments
- [ ] Break tight coupling between model / dataloader to save native model to central node (aggregator) at end of experiment

### PR DESCRIPTION
**Pull request by igor-davidyuk**
A new Python API was introduced. It allows defining Federated Learning tasks, data loaders, and models in an interactive environment such as Jupyter Notebook taking advantage of the introduced interactive API component.
Defined objects are then serialized and may be transferred to a remote node.

Fixes #13

### CODE REVIEW

**Comment by igor-davidyuk** [comment]
I have read the CLA Document and I hereby sign the CLA

File: openfl/interface/interactive_api/experiment.py [hunk 1]
```diff
@@ -0,0 +1,335 @@
+# We need assigner interface
+import os
+from copy import deepcopy
+from logging import getLogger
+import functools
+from collections import defaultdict
+from openfl.federated import Plan
+from pathlib import Path
+from importlib import resources
```
Comment (psfoley) [inline openfl/interface/interactive_api/experiment.py:9]: This breaks compatibility with python < 3.7. It would be good to investigate whether there's an alternative to this

File: openfl/interface/interactive_api/experiment.py [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+        )
+        return tensor_dict
+
+    def _prepare_plan(self, model_provider, task_keeper, data_loader,
+                      rounds_to_train,
+                      delta_updates=False, opt_treatment='RESET',
+                      model_interface_file='model_obj.pkl', tasks_interface_file='tasks_obj.pkl',
+                      dataloader_interface_file='loader_obj.pkl'):
+        # Load the default plan
+        with resources.path('openfl.interface.interactive_api', 'plan.yaml') as plan_path:
```
Comment (psfoley) [inline openfl/interface/interactive_api/experiment.py:123]: `plan.yaml` in the interactive_api folder does not get copied to this location during openfl installation. It would be good to either move this file to an artifact directory and rework `setup.py` so this gets copied properly, or have it reside in a default workspace directory under `openfl-workspace`. The advantage of the second approach is openfl-workspace continues to be the central location for all plan templates

File: openfl/interface/workspace.py [hunk 3]
```diff
@@ -108,34 +107,41 @@ def export_():
...
-                # 'fx create workspace' and current one.
-                echo(f'Writing {package}=={version} '
-                     f'to {requirements_filename}...')
-                f.write(f'{package}=={version}\n')
-            elif version is None:  # local dependency
-                warn(f'Could not generate requirements for {package}.'
-                     f' Consider installing it manually after workspace'
-                     f' import.')
-    echo(f'{export_requirements_filename} written.')
+    # requirements_filename = 'requirements.txt'
```
Comment (psfoley) [inline openfl/interface/workspace.py:110]: Does the removal of this break the existing export mechanism for the legacy API? If not, this commented out code should be removed

**Comment by igor-davidyuk** [comment]
I would like to make sure that metrics returning from tasks that are not declared in the plan are aggregated and treated correctly by the aggregator.

File: openfl/interface/workspace.py (hunk 3 above)
Comment (igor-davidyuk) [reply openfl/interface/workspace.py]: The new version works. Removed commented code

File: openfl/interface/interactive_api/experiment.py (hunk 2 above)
Comment (igor-davidyuk) [reply openfl/interface/interactive_api/experiment.py]: Moved default plan for the Interactive python API to `openfl/openfl-workspace/workspace/plan/plans/default/base_plan_interactive_api.yaml`

File: openfl/interface/interactive_api/experiment.py (hunk 1 above)
Comment (igor-davidyuk) [reply openfl/interface/interactive_api/experiment.py]: Done

**Comment by igor-davidyuk** [comment]
> I would like to make sure that metrics returning from tasks that are not declared in the plan are aggregated and treated correctly by the aggregator.

File: setup.py [hunk 4]
```diff
@@ -56,6 +60,7 @@
         'ipykernel',
         'flatten_json',
         'cryptography>=3.4.6',
+        'cloudpickle',
```
Comment (psfoley) [inline setup.py:63]: `dill` should probably be added here as well

Comment (igor-davidyuk) [reply setup.py:63]: I do not think we want to include all plugins dependencies into requirements.

**Review by alexey-gruzdev — APPROVED** [review]
